### DISCUSSION TITLE
google setting white bulb to not true white

### DISCUSSION THREAD

**Discussion by AalianKhan** [arendst/Tasmota#11956] [Q&A] [answered]
Hello,
I have tasmota 9.4 up and running on my techin SB50 lightbulb and it's working great.  
I have it connected to home assistant, which connects to google home so I can use my speakers.
When I say hey google change the light to white, it turns it this wired pinkish white (FFDDAC00) I have no idea why but I guess it is what it is. I couldn't find a way to tell tasmota to change the color when it changes to that color. Is this possible with rules? If so what is the trigger I should use? I tried `on Color#state do color FFFFFFFF endon` to test it out but Iit didn't work

**Comment by barbudor** [comment, marked as answer]
Hi 
What command do you see arriving in the console (may be with Weblog 3) when you are asking for "White" or other colors ?
Does the sliders on the web GUI works as expected ?

Beware that your rule may trigger a rule-loop.

**Reply by AalianKhan** [reply]
I see the following command
```stat/tasmota_9815D0/RESULT = {"POWER":"ON","Dimmer":100,"Color":"FFDDAC00","HSBColor":"35,33,100","White":0,"Channel":[100,87,67,0]}```
and
no the sliders don't change to pure white
![image](https://user-images.githubusercontent.com/39784853/116657984-4cc58d80-a95d-11eb-8ef3-757f74cc001d.png)

**Reply by AalianKhan** [reply]
the rule I tried was this
`on Color#state=FFDDAC00 do color FFFFFFFF endon`

**Reply by barbudor** [reply]
I meant, when you manually change the sliders on the GUI, does it behave as expected ?

It looks like the Tasmota is doing what Google is asking. You may need to teach right colors to Google.
I found quite a lot of people complaining of similar cases with different type of bulb (Philips, TP-Link, ...)
All seems to point to Google although I've not seen any recipe for solving it.

**Reply by AalianKhan** [reply]
yeah, it does work as expected when changing it to white from wither the GUI or home assistant. I was hoping for some sort of rule that will solve this issue.

**Reply by barbudor** [reply]
Your rule should work
May be try `on color#state do var9 %value% endon` to see if the rule is triggered and what you exactly receive for an accurate comparison

**Reply by sfromis** [reply]
Just to have the details clear, you did not post a *command* used to set the "white", only the `/RESULT` appearing. Try activating `Weblog 3` and see if you actually see a command or other output before that result message. There is more than one way to issue commands to affect a light, and details may make a difference to what would work best to do.

**Reply by AalianKhan** [reply]
Yes, I tried using `on color#state do power1 0 endon` before to test if its triggering but nothing happened and no related logs were found. 

@sfromis I can change the light to white and it works beautifully with home assistant too. The problem arises when I ask my google assistant to change the color to white, instead of the assistant publishing FFFFFFFF or something similar, it publishes a reddish tint of white. This doesn't turn on the white LEDs on my bulb thus being quite dim.

**Reply by sfromis** [reply]
And that's why it could be interesting to see any details on the arriving command (or whatever), instead of only the response from Tasmota.

**Reply by sfromis** [reply]
Instead of `on color#state do ...` try using `on color do ...`
(Worked on a bulb of mine)
